src/JWT.php

```diff
@@ -380,11 +448,17 @@ public static function jsonEncode(array $input): string
         }
         if ($errno = \json_last_error()) {
             self::handleJsonError($errno);
-        } elseif ($json === 'null') {
-            throw new DomainException('Null result with non-null input');
+        } elseif ($json === 'null' && $input !== null) {
```

Comment:
Oops, I merged this incorrectly.
```suggestion
        } elseif ($json === 'null') {
```